src/modules/settings/NetworkSettingsTab.tsx

```diff
@@ -37,10 +50,43 @@ export default function NetworkSettingsTab({ account }: Props) {
           setRoutingPeerDNSSetting(toggle);
           mutate("/accounts");
         }),
-      loadingMessage: "Saving the network settings...",
+      loadingMessage: "Updating DNS wildcard setting...",
     });
   };
 
+  const { hasChanges, updateRef } = useHasChanges([customDNSDomain]);
+
+  const saveChanges = async () => {
+    notify({
+      title: "Custom DNS Domain",
+      description: `Custom DNS Domain successfully updated.`,
+      promise: saveRequest
+        .put({
+          id: account.id,
+          settings: {
+            ...account.settings,
+            dns_domain: customDNSDomain || "",
+          },
+        })
+        .then(() => {
+          mutate("/accounts");
+          updateRef([customDNSDomain]);
+        }),
+      loadingMessage: "Updating Custom DNS domain...",
+    });
+  };
+
+  const domainError = useMemo(() => {
+    if (customDNSDomain == "") return "";
+    const valid = validator.isValidDomain(customDNSDomain, {
+      allowWildcard: false,
+      allowOnlyTld: false,
```

Comment:
```suggestion
      allowOnlyTld: true,
```

This should be allowed, no? E.g. if someone wants `peer1.netbird` or something like that

Reply:
@heisbrot This ^^. Why not allow TLD-only custom domains?

Reply:
@alindt Currently, that's a backend limitation. We will fix it once the backend is ready.